tensorflow/compiler/xla/service/gpu/BUILD

```diff
@@ -1049,28 +1066,44 @@ cc_library(
         "//tensorflow/compiler/xla/service:zero_sized_hlo_elimination",
         "//tensorflow/compiler/xla/service/gpu/llvm_gpu_backend",
         "//tensorflow/compiler/xla/service/llvm_ir:llvm_util",
-        "//tensorflow/core:cuda_libdevice_path",
         "//tensorflow/core:lib",
         "//tensorflow/core:lib_internal",
         "//tensorflow/core:regexp_internal",
         "//tensorflow/core:stream_executor_no_cuda",
         "//tensorflow/core/profiler/lib:traceme",
         "//tensorflow/stream_executor:stream_executor_headers",
-        "//tensorflow/stream_executor/cuda:cuda_diagnostics",
-        "//tensorflow/stream_executor/cuda:ptxas_utils",
         "@com_google_absl//absl/container:node_hash_map",
         "@com_google_absl//absl/memory",
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/types:optional",
         "@com_google_absl//absl/types:span",
         "@llvm//:core",
-    ],
+    ] + if_cuda_is_configured([
+        ":cudnn_conv_algorithm_picker",
+        ":cudnn_conv_pad_for_tensor_cores",
+        ":cudnn_fused_conv_rewriter",
+        ":cusolver_rewriter",
+        ":gemm_algorithm_picker",
+        ":gemm_rewriter",
+        "//tensorflow/core:cuda_libdevice_path",
+        "//tensorflow/stream_executor/cuda:cuda_diagnostics",
+        "//tensorflow/stream_executor/cuda:ptxas_utils",
+    ]) + if_rocm_is_configured([
+        # ROCM TODO: Enable these after pending PRs get merged.
+        #":miopen_conv_algorithm_picker",
+        #"//tensorflow/core:rocm_rocdl_path",
+    ]),
 )
 
 cc_library(
-    name = "nvptx_compiler",
-    srcs = ["nvptx_compiler_registration.cc"],
-    deps = [":nvptx_compiler_impl"],
+    name = "gpu_compiler",
```

Comment:
Did you consider making `nvptx_compiler` and `amdgpu_compiler` separate cc_libraries? You can pick the one you want, similarly to picking the right stream_executor in `gpu_plugin` above.
This would make the BUILD file easier to read and maintain and be consistent with the way we specialize the stream_executor target for cuda and rocm.

Reply:
ok, let me separate `gpu_compiler_impl` into `nvptx_compiler` and `amdgpu_compiler`, and let `gpu_compiler` pick the right one based on bazel configuration.